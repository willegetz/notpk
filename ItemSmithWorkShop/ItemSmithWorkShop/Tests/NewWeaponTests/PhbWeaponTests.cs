using System.Text;
using ApprovalTests;
using ItemSmithWorkShop.Weapons;
using ItemSmithWorkShop.Weapons.MaterialTypes;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using ItemSmithWorkShop.Weapons.MagicEnchantments;
using System;
using System.Collections.Generic;

namespace ItemSmithWorkShop.Tests.NewWeaponTests
{
	[TestClass]
	public class PhbWeaponTests
	{
		[TestMethod]
		public void TestDisplayDagger()
		{
			var dagger = new PhbWeapon("Dagger");
			Approvals.Verify(dagger.ToString());
		}

		[TestMethod]
		public void TestDisplayMithral()
		{
			var mithral = new Mithral();
			Approvals.Verify(mithral.ToString());
		}

		[TestMethod]
		public void TestDisplayAnarchicEnchantment()
		{
			var anarchic = new WeaponEnchantment("Anarchic");
			Approvals.Verify(anarchic.ToString());
		}

		[TestMethod]
		public void TestDisplayFlamingBurstEnchantment()
		{
			var flameBurst = new WeaponEnchantment("Flaming Burst");
			Approvals.Verify(flameBurst.ToString());
		}
	}
}
