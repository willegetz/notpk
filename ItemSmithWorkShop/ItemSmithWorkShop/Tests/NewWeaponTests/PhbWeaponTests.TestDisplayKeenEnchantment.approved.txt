Enchantment: 'Keen'
Affix: 'Pre'
Cost Modifier: '+1'
Weapon Usage: 'Melee, Ranged'
Standard Damage Bonus: ''
Threat Range Modifier: '2'
Critical Damage Bonus: 'False'
Damage Type: ''
Magic Aura: 'Moderate transmutation'
Minimum Caster Level: '10'
Range Increment Multiplier: '0'
Required Feats: 'Craft Magic Arms and Armor'
Required Spells: 'keen edge'
Additional Requirements: ''
Notes: 'Only piercing or slashing weapons can be keen. The keen
	benefit does not stack with any other effect that expands
	the threat range of a weapon.'